server/routes/appointments/create-and-edit/handlers/schedule.ts

```diff
@@ -2,56 +2,79 @@ import { Request, Response } from 'express'
 import { plainToInstance } from 'class-transformer'
 import { YesNo } from '../../../../@types/activities'
 import ActivitiesService from '../../../../services/activitiesService'
+import EditAppointmentService from '../../../../services/editAppointmentService'
 import SimpleDate from '../../../../commonValidationTypes/simpleDate'
-import { AppointmentType } from '../appointmentJourney'
+import { AppointmentJourneyMode, AppointmentType } from '../appointmentJourney'
 
 export default class ScheduleRoutes {
-  constructor(private readonly activitiesService: ActivitiesService) {}
+  constructor(
+    private readonly activitiesService: ActivitiesService,
+    private readonly editAppointmentService: EditAppointmentService,
+  ) {}
 
   GET = async (req: Request, res: Response): Promise<void> => {
     const { user } = res.locals
-    const { appointmentJourney, bulkAppointmentJourney } = req.session
+    const { occurrenceId } = req.params
+    const { appointmentJourney, bulkAppointmentJourney, editAppointmentJourney } = req.session
     const { preserveHistory } = req.query
 
     let backLinkHref: string
-    if (req.session.appointmentJourney.type === AppointmentType.BULK) {
+    if (req.session.appointmentJourney.mode === AppointmentJourneyMode.EDIT) {
+      backLinkHref = 'date-and-time'
+      if (editAppointmentJourney?.addPrisoners) {
+        backLinkHref = 'prisoners/add/review-prisoners'
+      }
+    } else if (req.session.appointmentJourney.type === AppointmentType.BULK) {
       backLinkHref = 'review-bulk-appointment'
     } else {
       backLinkHref = req.session.appointmentJourney.repeat === YesNo.YES ? 'repeat-period-and-count' : 'repeat'
     }
 
-    const prisonNumbers =
-      appointmentJourney.type === AppointmentType.BULK
-        ? bulkAppointmentJourney.appointments.map(a => a.prisoner.number)
-        : appointmentJourney.prisoners.map(p => p.number)
+    let prisonNumbers
+    if (appointmentJourney.type === AppointmentType.BULK) {
+      prisonNumbers = bulkAppointmentJourney.appointments.map(a => a.prisoner.number)
+    } else if (editAppointmentJourney?.addPrisoners) {
+      prisonNumbers = editAppointmentJourney.addPrisoners.map(p => p.number)
+    } else {
+      prisonNumbers = appointmentJourney.prisoners.map(p => p.number)
+    }
+
+    const appointmentStartDate = editAppointmentJourney?.startDate ?? appointmentJourney.startDate
 
     const scheduledEvents = await this.activitiesService
       .getScheduledEventsForPrisoners(
-        plainToInstance(SimpleDate, appointmentJourney.startDate).toRichDate(),
+        plainToInstance(SimpleDate, appointmentStartDate).toRichDate(),
         prisonNumbers,
         user,
       )
       .then(response => [
         ...response.activities,
-        ...response.appointments,
+        ...response.appointments.filter(e => e.appointmentOccurrenceId !== +occurrenceId),
```

Comment:
I don't think there's a test that covers filtering out the appointment being edited from the prisoner's schedule. Worth adding one?